libvast/src/view.cpp

```diff
@@ -41,30 +43,54 @@ bool operator==(subnet_view x, subnet_view y) noexcept {
   return x.network_ == y.network_ && x.length_ == y.length_;
 }
 
-vector_view::iterator vector_view::begin() const {
-  // TODO: const_cast okay?
-  return {vector_view_ptr{const_cast<vector_view*>(this), true}, 0};
+
+default_vector_view::default_vector_view(const vector& xs) : xs_{xs} {
```

Comment:
This constructor has a sink argument, I would use `default_vector_view(vector xs) : xs_{std::move(xs)} ... ` instead.

Reply:
I'm not sure what you mean. This is a *view* of a `vector`, it should neither copy nor own the instance. That's why `vector_` is a const reference.